### PR TITLE
Introduced the STORED parameter to change ECL scritps at run time. Also

### PR DESCRIPTION
**Pull request by jpmaughan**
reformatted the way we create the ECL command line to be made form lists. Also introduced the option to include STORED variables in run_ecl_script

### CODE REVIEW

**Comment by jpmaughan** [comment]
Fixes #39 #38 #74

File: hpycc/connection.py [hunk 1]
```diff
@@ -191,6 +192,9 @@ def run_ecl_script(self, script, syntax_check, delete_workunit):
             executed.
         delete_workunit: bool
             Delete workunit once completed.
+        stored : dict
```
Comment (cqkh42) [inline hpycc/connection.py]: This should be dict or None

File: hpycc/connection.py [hunk 2]
```diff
@@ -256,7 +271,7 @@ def run_url_request(self, url, max_attempts, max_sleep):
 
         Returns
         -------
-        r: requests.models.Response
+        r: requests.models.
```
Comment (cqkh42) [inline hpycc/connection.py]: did you mean to delete this?

File: hpycc/connection.py [hunk 3]
```diff
@@ -342,6 +357,11 @@ def run_ecl_string(self, string, syntax_check, delete_workunit):
             executed.
         delete_workunit: bool
             Delete workunit once completed.
+        stored : dict
+            Dictionary of key value pairs do to replace stored
```
Comment (cqkh42) [inline hpycc/connection.py]: there is a typo here. Should also say dict or None. A few docstrings also need optional adding. Let's make sure the docstring is standardised

File: tests/test_with_server/test_connection.py [hunk 4]
```diff
@@ -210,13 +216,13 @@ def test_run_ecl_string_returns_result_of_run_ecl_script(self):
...
         mock.assert_called()
 
     @patch.object(hpycc.connection.delete, "delete_workunit")
     def test_get_run_ecl_string_doesnt_run_delete_workunit_if_false(
             self, mock):
         script = "OUTPUT(2);"
         conn = hpycc.Connection("user", test_conn=False)
-        conn.run_ecl_string(script, False, False)
+        conn.run_ecl_string(script, False, False, {})
         self.assertFalse(mock.called)
```
Comment (cqkh42) [inline tests/test_with_server/test_connection.py]: we need tests for the new behaviour in run_ecl_script and run_ecl_string

File: tests/test_with_server/test_get.py [hunk 5]
```diff
@@ -185,6 +185,44 @@ def test_get_output_doesnt_run_delete_workunit_if_false(self, mock):
         _get_output_from_ecl_string(self.conn, script, delete_workunit=False)
         self.assertFalse(mock.called)
 
+    def test_get_output_stored_variables_change_output_same_type_string(self):
```
Comment (cqkh42) [inline tests/test_with_server/test_get.py]: this doesn;t actually test the types. it tests our parsing of the tyoes as they come back out. try having a number, and output 1 + a, then try a as a string, etc.

File: tests/test_with_server/test_run.py [hunk 6]
```diff
@@ -73,3 +73,31 @@ def test_run_script_returns_true(self):
...
+
+    def test_run_script_does_nothing_when_not_using_stored_(self):
+        conn = hpycc.Connection("user", test_conn=False)
+        good_script = "a:='b':STORED('a');a;"
+
+        with TemporaryDirectory() as d:
+            p = os.path.join(d, "test.ecl")
+            with open(p, "w+") as file:
+                file.write(good_script)
+            res = conn.run_ecl_script(p, syntax_check=True,
```
Comment (cqkh42) [inline tests/test_with_server/test_run.py]: these tests don;t actually test run!!

**Comment by cqkh42** [comment]
fixes #74 #39 #38

**Pull request merged**